Sources/TripKitUI/views/trip overview/TKUISegmentCellHelper.swift

```diff
@@ -28,10 +28,12 @@ enum TKUISegmentCellHelper {
     button.accessibilityLabel = action.accessibilityLabel
     button.rx.tap
       .subscribe(onNext: { [unowned card] in
-        let update = action.handler(action, card, model, button)
-        if update {
-          button.setTitle(action.title, for: .normal)
-          button.accessibilityLabel = action.accessibilityLabel
+        DispatchQueue.main.async {
```

Comment:
any reason why this is used instead of the rx usage of `.observeOn(...)`?

Reply:
The Swift compiler wants a compile time guarantee for the block to run on the main queue, which `observe(on:)` does not provide.